Yoast/javascript
# 2547 make prominent word naming clearer

## Summary


This PR can be summarized in the following changelog entry:

* Refactors the code to use another term than 'prominent words', 'keyword suggestions' or 'relevant words'.

## Relevant technical choices:

* I boyscouted the `WordOccurrencesWrapper` class in the components app. I added comments and renamed some variables, methods and functions.

## Test instructions

This PR can be tested by following these steps:

**Note**: this PR needs to be tested in unison with the premium PR: https://github.com/Yoast/wordpress-seo-premium/pull/2613

* Check if the readability- and SEO-analysis work as expected.
* Check if the word occurrences insight works as expected.
* Check if the internal linking suggestions work as expected.
* Check if site-wide internal linking works as expected.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes #https://github.com/Yoast/wordpress-seo-premium/issues/2547
